akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowGroupBySpec.scala

```diff
@@ -336,6 +336,18 @@ class FlowGroupBySpec extends StreamSpec {
       s1.expectError(ex)
     }
 
+    "resume when exceeding maxSubstreams" in {
+      val (up, down) = Flow[Int]
+        .groupBy(0, identity).mergeSubstreams
+        .withAttributes(ActorAttributes.supervisionStrategy(resumingDecider))
+        .runWith(TestSource.probe[Int], TestSink.probe)
+
+      down.request(1)
+
+      up.sendNext(1)
+      down.expectNoMessage(1.second)
```

Comment:
This doesn't verify that it resumes though? Shouldn't it make sure to hit the max (not being 0 then) and then make sure an element passes through?

Reply:
I can look further into this, but I'll wait to hear about the outcome from other concerns first. :-)

Reply:
I have no concerns except for the two test comments, and Patrik gave a LGTM, so please do look into it thanks!

Reply:
Actually, wouldn't that be an expectation of the existing set of tests?

Reply:
i.e. maybe I should rename this test as there's already one for resumption, I think (I've not looked though)

Reply:
It looked like it was added by this PR but I see it is already in master. Fixing it to actually test what it says would be good though.